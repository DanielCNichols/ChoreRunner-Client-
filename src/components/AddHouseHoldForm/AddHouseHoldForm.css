.add-household-form {
    display: flex;
    flex-direction: column;
    justify-content: center;
    align-items: center;
    margin: 20px auto;
  }

.add-household-form fieldset {
    border: none;
    display: flex;
    flex-direction: column;
}

.add-household-form legend {
    font-family: "Press Start 2P", italic;
    margin: 10px auto;
    color: #FFD900;

}

.add-household-form button {
    text-align: center;
    margin: 10px auto 20px auto;
    font-family: 'Electrolize', sans-serif;
    font-size: 20px;
    padding: 10px;
    background-color: rgb(255, 217, 0);
    text-shadow: -1px -1px rgb(105, 90, 0), 1px 1px white;
    color: #725f00;
    -webkit-border-radius: 7px;
    -moz-border-radius: 7px;
    -o-border-radius: 7px;
    border-radius: 7px;
    box-shadow: 0 0.3em rgb(134, 112, 0);
    cursor: pointer;
  }

.add-household-form label {
    font-family: "Electrolize", sans-serif;
    margin: 0 auto;
    padding: 0;
    font-size: 1.3em;
}

.add-household-form input {
    margin: 10px auto 20px auto;
    
}



@media screen and (min-width: 800px) {
    .add-household-form {
        display: flex;
        flex-direction: column;
        justify-content: center;
        align-items: center;
      }
}
  