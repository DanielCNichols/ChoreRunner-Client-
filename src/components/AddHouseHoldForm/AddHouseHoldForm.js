import React, { useState } from 'react';
import s from './AddHouseHoldForm.module.css';
import {
  Input,
  FormElement,
  Fieldset,
  Label,
  Legend,
} from '../../components/Form/Form';
import ApiService from '../../services/api-service';

export default function AddHouseHoldForm({ handleAdd, toggleAdd }) {
  const [groupName, setGroupName] = useState('');

  const [confirm, setConfirm] = useState('');

  const [error, setError] = useState({
    groupName: null,
    server: null,
  });

  const handleInputChange = ev => {
    ev.persist();
    setGroupName(ev.target.value);
  };

  const handleHouseholdSubmit = async ev => {
    try {
      ev.preventDefault();
      setError({ groupName: null });

      if (!groupName) {
        return setError({ groupName: 'Group name is required' });
      }

      let added = await ApiService.postHousehold(groupName);

      handleAdd(added);
      toggleAdd();
    } catch (error) {
      setError({ error: error });
    }
  };

  return (
    <form className={s.addHouseholdForm} onSubmit={handleHouseholdSubmit}>
      <Fieldset>
        <Legend>Add a Group</Legend>
        <FormElement>
          <Label htmlFor="groupName">Group Name</Label>
          {error.groupName && (
            <div className={s.error}>
              <span>{error.groupName}</span>
            </div>
          )}
          <Input
            onChange={handleInputChange}
            name="groupName"
            type="text"
            value={groupName}
          />
        </FormElement>

        {error.server && (
          <div className={s.error}>
            <span>{error.server}</span>
          </div>
        )}
        <div className={s.formButtons}>
          <button className="arcadeButton" type="button" onClick={toggleAdd}>
            Cancel
          </button>
          <button className="arcadeButton" type="submit">
            Create
          </button>
        </div>
      </Fieldset>
    </form>
  );
}
