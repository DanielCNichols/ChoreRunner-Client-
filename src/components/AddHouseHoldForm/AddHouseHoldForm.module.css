@import '../../vars.css';

.error {
  color: red;
  font-weight: bold;
  margin-bottom: 5px;
}

.addHouseholdForm {
  width: 100%;
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
  margin: 20px auto;
  background-color: var(--color-menus);
  padding: 20px 20px;
  border: 2px solid white;
  border-radius: 6px;
}

.addHouseholdForm fieldset {
  width: 80%;
  min-width: 245px;
  border: none;
  display: flex;
  flex-direction: column;
}

.addHouseholdForm legend {
  font-family: 'Press Start 2P', italic;
  margin: 10px auto;
  color: #ffd900;
}

.addHouseholdForm label {
  color: white;
  margin-bottom: 5px;
  text-align: left;
  font-family: var(--font-body);
  font-size: 1em;
}

.formButtons {
  width: 100%;
  display: flex;
  justify-content: space-between;
}

.addHouseholdForm button {
  width: 100px;
  height: 45px;
  text-align: center;
  margin: 10px auto;
  font-family: 'Press Start 2P', cursive;
  font-size: 0.8em;
  padding: 7px;
  background-color: rgb(255, 217, 0);
  text-shadow: -1px -1px rgb(105, 90, 0), 1px 1px white;
  color: #725f00;
  -webkit-border-radius: 7px;
  -moz-border-radius: 7px;
  -o-border-radius: 7px;
  border-radius: 7px;
  box-shadow: 0 0.3em rgb(134, 112, 0);
  cursor: pointer;
}

/* 
.add-household-form button {
  text-align: center;
  margin: 10px auto 20px auto;
  font-family: 'Electrolize', sans-serif;
  font-size: 20px;
  padding: 10px;
  background-color: rgb(255, 217, 0);
  text-shadow: -1px -1px rgb(105, 90, 0), 1px 1px white;
  color: #725f00;
  -webkit-border-radius: 7px;
  -moz-border-radius: 7px;
  -o-border-radius: 7px;
  border-radius: 7px;
  box-shadow: 0 0.3em rgb(134, 112, 0);
  cursor: pointer;
} */
