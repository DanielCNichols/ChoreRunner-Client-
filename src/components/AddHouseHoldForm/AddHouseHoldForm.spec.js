import React from 'react';
import ReactDOM from 'react-dom';
import AddHouseHoldForm from './AddHouseHoldForm';
import renderer from 'react-test-renderer';
import { expectation } from 'sinon';

describe('AddHouseholdForm tests', () => {
  let handleAdd = () => {};
  let toggleAdd = () => {};

  it('renders without crashing', () => {
    const div = document.createElement('div');

    ReactDOM.render(
      <AddHouseHoldForm toggleAdd={toggleAdd} handleAdd={handleAdd} />,
      div
    );

    ReactDOM.unmountComponentAtNode(div);
  });

  it('renders UI as expected', () => {
    const tree = renderer
      .create(<AddHouseHoldForm toggleAdd={toggleAdd} handleAdd={handleAdd} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
