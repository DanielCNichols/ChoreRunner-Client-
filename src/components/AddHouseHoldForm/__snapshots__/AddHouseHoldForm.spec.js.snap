// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddHouseholdForm tests renders UI as expected 1`] = `
<form
  className="addHouseholdForm"
  onSubmit={[Function]}
>
  <fieldset
    className="Fieldset"
  >
    <legend
      className="Legend"
    >
      Add a Group
    </legend>
    <div
      className="FormElement"
    >
      <label
        className="Label"
        htmlFor="groupName"
      >
        Group Name
      </label>
      <input
        className="Input"
        name="groupName"
        onChange={[Function]}
        type="text"
        value=""
      />
    </div>
    <div
      className="formButtons"
    >
      <button
        className="arcadeButton"
        onClick={[Function]}
        type="button"
      >
        Cancel
      </button>
      <button
        className="arcadeButton"
        type="submit"
      >
        Create
      </button>
    </div>
  </fieldset>
</form>
`;
